src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -525,6 +527,16 @@ private boolean isContainVerifiableType(DetailAST modifiers) {
         return result;
     }
 
+    /**
+     * Checks a record component definition. Record components are private
+     * final fields, so we do not consider memberModifiers field.
+     *
+     * @param recordComponentDef the record component definition to check.
+     */
+    public void visitRecordComponentDef(DetailAST recordComponentDef) {
+        checkClassName(recordComponentDef);
```

Comment:
Simply use this method in "case"

Reply:
So I should delete this method here and update it in case?

Reply:
But in case we are using (DetailAST ast) so how can I use (DetailASR recordComponentDef) in CASE?

Reply:
@romani sir, Can we use this condition?
```
public void visitRecordComponentDef(DetailAST recordComponentDef) {
        if (recordComponentDef.getType() == TokenTypes.RECORD_COMPONENT_DEF) {
            checkClassName(recordComponentDef);
        }
```

Reply:
```
case TokenTypes.RECORD_COMPONENT_DEF:
                checkClassName(ast);
```

Reply:
Thanks Sir!
Should I delete this method now?

Reply:
Done!